### PR TITLE
Fix #5869: Increased tap area of Switch button by adding additional clear button for ease of use

### LINKED ISSUE #5869: The `Switch` button from the toast notification is too small
## Steps to reproduce
1. Launch Firefox.
2. Open any website.
3. Long tap on a link from the website.
4. Tap on the `Open in new tab`.

 ### Expected behavior
The `Switch` button from the toast notification should be bigger and have a better position.
### Actual behavior
The `Switch` button from the toast notification is too small.
### Device information

* iOS device: **iPhone Xs Max(iOS 12.4), iPhone Xs(iOS 13.2.2)**
* Firefox version: **v21.0(16860)**

### Notes
Note that sometimes the users can tap by mistake a little bit near the button and they will not be redirected to the page.
The toast can look better and having bigger buttons when the bottom bar is hidden.
Also, the `New tab oepend` text can be resized.
unnamed (3)

unnamed (4) - Copy

unnamed

### PR DESCRIPTION
**Pull request by nbhasin2**
The idea here is to add more area for user to be able to tap on the "Switch" button when they open a new tab.

### CODE REVIEW

File: Client/Frontend/Browser/ButtonToast.swift [hunk 1]
```diff
@@ -133,16 +124,54 @@ class ButtonToast: Toast {
...
         }
-
         return toastView
     }
-
-    @objc func buttonPressed(_ gestureRecognizer: UIGestureRecognizer) {
-        completionHandler?(true)
-        dismiss(true)
+    
+    func setupPaddedButton(stackView: UIStackView, buttonText:String?) {
```
Comment (garvankeeley) [inline Client/Frontend/Browser/ButtonToast.swift]: Not sure why `buttonText` is optional, wouldn't the button be empty without text?

File: Client/Frontend/Browser/ButtonToast.swift [hunk 2]
```diff
@@ -133,16 +124,54 @@ class ButtonToast: Toast {
...
+            paddedView.addSubview(roundedButton)
+            roundedButton.snp.makeConstraints { make in
+                make.height.equalTo(roundedButton.titleLabel!.intrinsicContentSize.height + 2 * ButtonToastUX.ToastButtonPadding)
+                make.width.equalTo(roundedButton.titleLabel!.intrinsicContentSize.width + 2 * ButtonToastUX.ToastButtonPadding)
+                make.centerY.equalTo(stackView.snp.centerY)
+                make.trailing.equalTo(paddedView.snp.trailing)
+            }
+            roundedButton.addGestureRecognizer(UITapGestureRecognizer(target: self, action: #selector(buttonPressed)))
+            
+            paddedView.snp.makeConstraints { make in
```
Comment (garvankeeley) [inline Client/Frontend/Browser/ButtonToast.swift]: A single `paddedView.makeConstraints` would be clearer

File: Client/Frontend/Browser/ButtonToast.swift [hunk 3]
```diff
@@ -43,6 +50,7 @@ class ButtonToast: Toast {
         self.snp.makeConstraints { make in
             make.height.greaterThanOrEqualTo(ButtonToastUX.ToastHeight)
         }
+
```
Comment (garvankeeley) [inline Client/Frontend/Browser/ButtonToast.swift]: nit: space

File: Client/Frontend/Browser/ButtonToast.swift [hunk 4]
```diff
@@ -60,6 +68,7 @@ class ButtonToast: Toast {
             icon.tintColor = UIColor.Photon.White100
             horizontalStackView.addArrangedSubview(icon)
         }
+
```
Comment (garvankeeley) [inline Client/Frontend/Browser/ButtonToast.swift]: nit: space

File: Client/Frontend/Browser/ButtonToast.swift (hunk 1 above)
Comment (nbhasin2) [reply Client/Frontend/Browser/ButtonToast.swift]: Refactored the code, so now just like before we don't add the button if there is no button text.

**Pull request merged**